x/staking/keeper/slash.go

```diff
@@ -232,19 +235,6 @@ func (k Keeper) slashRedelegation(ctx sdk.Context, validator types.Validator, re
 		slashAmount := slashAmountDec.TruncateInt()
 		totalSlashAmount = totalSlashAmount.Add(slashAmount)
 
-		// Don't slash more tokens than held
-		// Possible since the redelegation may already
-		// have been slashed, and slash amounts are calculated
-		// according to stake held at time of infraction
-		redelegationSlashAmount := sdk.MinInt(slashAmount, entry.Balance.Amount)
-
-		// Update entry if necessary
-		if !redelegationSlashAmount.IsZero() {
-			entry.Balance.Amount = entry.Balance.Amount.Sub(redelegationSlashAmount)
-			redelegation.Entries[i] = entry
-			k.SetRedelegation(ctx, redelegation)
-		}
```

Comment:
Could you explain or reference why this is removed?

Reply:
We never actually use this `Balance` anywhere, since we separately check that we don't burn too many shares.

Reply:
k yeah so SrcShares and Balance are both removed from RedelegationEntry struct